src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,78 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * Make this class immutable. See requirements in the README file
  */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        this.wheels = cloneWheels(wheels);
+        this.engine = engine == null ? null : engine.clone();
+    }
+
+    @Override
+    public boolean equals(Object thatCame) {
+        if (this == thatCame) {
+            return true;
+        }
+        if (thatCame == null || getClass() != thatCame.getClass()) {
+            return false;
+        }
+        Car car = (Car) thatCame;
+        return year == car.year && Objects.equals(color, car.color)
+                && Objects.equals(wheels, car.wheels) && Objects.equals(engine, car.engine);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(year, color, wheels, engine);
+    }
+
+    public int getYear() {
+        return year;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public List<Wheel> getWheels() {
+        return new ArrayList<>(wheels);
```

Comment:
```suggestion
        return cloneWheels(wheels);
```
you've created new list but with same wheels